StreamChatUITestsApp/StreamChat/ChannelListVC.swift

```diff
@@ -0,0 +1,18 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import StreamChat
+import StreamChatUI
+
+final class ChannelListVC: ChatChannelListVC {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+        
+        let query = ChannelListQuery.init(filter: .containMembers(userIds: [ChatClient.shared.currentUserId!]))
+        self.controller = ChatClient.shared.channelListController(query: query)
```

Comment:
That seems like a miss-use 🙂

In `super.viewDidLoad` custom VC lifecycle methods are triggered (`setUp/etc.`) where `controller` property is accessed. Since we assign `controller` after `super.viewDidLoad` it would crash because in `setUp` a controller would be nil. However, it's not crashing because `controller` is already assigned in `StreamChatWrapper.makeChannelListViewController` but here we create & assign another LLC controller instance.

The right way would be to remove this custom subclass and user factory method `ChatChannelListVC.make` in `makeChannelListViewController` that ensures `controller` is assigned before the VC is shown:

```swift
func makeChannelListViewController() -> UIViewController {
        // UI
        let query = ChannelListQuery(filter: .containMembers(userIds: [userCredentials?.id ?? ""]))
        let controller = ChatClient.shared.channelListController(query: query)
        
        return ChatChannelListVC.make(with: controller)
    }
```

Reply:
Thanks for the insight. Done :)